frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx

```diff
@@ -1803,78 +1853,15 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
-                interval,
-                device_tab,
-                source_tab,
-                graphs_tab,
-                path_tab,
-                geography_tab,
-                path_cleaning,
-                filter_test_accounts,
-                compare_filter,
-            }
-        ) => {
-            const parsedFilters = isWebAnalyticsPropertyFilters(filters) ? filters : undefined
-
-            if (parsedFilters && !objectsEqual(parsedFilters, values.webAnalyticsFilters)) {
-                actions.setWebAnalyticsFilters(parsedFilters)
-            }
-            if (
-                conversionGoalActionId &&
-                conversionGoalActionId !== (values.conversionGoal as ActionConversionGoal)?.actionId
-            ) {
-                actions.setConversionGoal({ actionId: parseInt(conversionGoalActionId, 10) })
-            } else if (
-                conversionGoalCustomEventName &&
-                conversionGoalCustomEventName !== (values.conversionGoal as CustomEventConversionGoal)?.customEventName
-            ) {
-                actions.setConversionGoal({ customEventName: conversionGoalCustomEventName })
-            }
-            if (
-                (date_from && date_from !== values.dateFilter.dateFrom) ||
-                (date_to && date_to !== values.dateFilter.dateTo) ||
-                (interval && interval !== values.dateFilter.interval)
-            ) {
-                actions.setDatesAndInterval(date_from, date_to, interval)
-            }
-            if (device_tab && device_tab !== values._deviceTab) {
-                actions.setDeviceTab(device_tab)
-            }
-            if (source_tab && source_tab !== values._sourceTab) {
-                actions.setSourceTab(source_tab)
-            }
-            if (graphs_tab && graphs_tab !== values._graphsTab) {
-                actions.setGraphsTab(graphs_tab)
-            }
-            if (path_tab && path_tab !== values._pathTab) {
-                actions.setPathTab(path_tab)
-            }
-            if (geography_tab && geography_tab !== values._geographyTab) {
-                actions.setGeographyTab(geography_tab)
-            }
-            if (path_cleaning && path_cleaning !== values.isPathCleaningEnabled) {
-                actions.setIsPathCleaningEnabled([true, 'true', 1, '1'].includes(path_cleaning))
-            }
-            if (filter_test_accounts && filter_test_accounts !== values.shouldFilterTestAccounts) {
-                actions.setShouldFilterTestAccounts([true, 'true', 1, '1'].includes(filter_test_accounts))
-            }
-            if (compare_filter && !objectsEqual(compare_filter, values.compareFilter)) {
-                actions.setCompareFilter(compare_filter)
-            }
-        },
+        '/web': (pathParams, queryParams) => setFilters(actions, values, pathParams, queryParams),
+        '/web/:productTab': (pathParams, queryParams) => setFilters(actions, values, pathParams, queryParams),
```

Comment:
I couldn't get this to work with a single string. I tried all sorts of regexes, but I don't think `kea` is that powerful. @mariusandra if you're keen to help here :)

Reply:
I'm not sure what you're trying to achieve, and Kea is as powerful as the user 😅. `urlToAction` takes a function that returns an object, so feel free to add whatever rows into the object that you need to.

```ts
    urlToAction(({ actions, values }) => {
        const obj = {}
        const handler = (path, search, hash) => {...}
        for (...) { obj[`/web/${product.name}`] => handler }
        return obj
    })
```

Reply:
Oh, of course, it's a function, I can just create the keys programatically, thanks